index.mjs

```diff
@@ -69,14 +78,23 @@ export function $(pieces, ...args) {
 }
 
 $.verbose = true
-// Try `command`, should cover all Bourne-like shells.
-// Try `which`, should cover most other cases.
-// Try `type` command, if the rest fails.
-$.shell = `${execSync('command -v bash || which bash || type -p bash')}`.trim()
-$.prefix = 'set -euo pipefail;'
-$.quote = shq
 $.cwd = undefined
 
+$.quote = (s) => s
+$.prefix = ''
+if ($.shell = which.sync('bash', { nothrow: true })) {
+  $.prefix = 'set -euo pipefail;'
+  $.quote = shq
+} else if ($.shell = whichSyncFallback(['pwsh', 'powershell'])) {
+  // Powershell is found on Windows. It has the required support for -c, unlike
+  // cmd.
+  // Prefer 'pwsh', which points to a newer version, for a higher likelihood of
+  // getting "correct" argument passing (PSNativeCommandArgumentPassing).
+  console.warn("Using powershell: no built-in quoting available yet.")
```

Comment:
As for this "yet"... I do have a quoting function available, but I am guessing you don't like putting it in here:

```js
(arg) => `'${arg.replace('\x00', '').replace(/'/g, "''")}'`
```